### PR TITLE
Improve trip reminder notification

### PR DESCRIPTION
**Pull request by binh-dam-ibigroup**
### Checklist

- [x] Appropriate branch selected _(all PRs must first be merged to `dev` before they can be merged to `master`)_
- [x] Any modified or new methods or classes have helpful JavaDoc and code is thoroughly commented
- [na] The description lists all applicable issues this PR seeks to resolve
- [na] The description lists any configuration setting(s) that differ from the default settings
- [x] All tests and CI builds passing

### Description

This PR makes the following improvements to the initial reminder notification:
- That notification now always appear first
- The portion reading "We will let you know if anything changes" is only shown if there are no other notifications.

### CODE REVIEW

File: src/main/java/org/opentripplanner/middleware/models/TripMonitorNotification.java [hunk 1]
```diff
@@ -85,23 +98,25 @@ public static TripMonitorNotification createDelayNotification(
...
-        TripMonitorNotification notification = new TripMonitorNotification();
-        notification.type = NotificationType.ITINERARY_NOT_FOUND;
-        notification.body = stillPossibleOnOtherMonitoredDaysOfTheWeek
-            ? "Your itinerary was not found in trip planner results for today! Please check realtime conditions and plan a new trip."
-            : "Your itinerary is no longer possible any monitored day of the week! Please plan and save a new trip.";
-        return notification;
+        return new TripMonitorNotification(
+            NotificationType.ITINERARY_NOT_FOUND,
+            stillPossibleOnOtherMonitoredDaysOfTheWeek
+                ? "Your itinerary was not found in trip planner results for today! Please check realtime conditions and plan a new trip."
```
Comment (AdrianaCeric) [inline src/main/java/org/opentripplanner/middleware/models/TripMonitorNotification.java]:
```suggestion
                ? "Your itinerary was not found in today's trip planner results. Please check real-time conditions and plan a new trip."
```

File: src/main/java/org/opentripplanner/middleware/models/TripMonitorNotification.java [hunk 2]
```diff
@@ -85,23 +98,25 @@ public static TripMonitorNotification createDelayNotification(
...
+                : "Your itinerary is no longer possible any monitored day of the week! Please plan and save a new trip."
```
Comment (AdrianaCeric) [inline src/main/java/org/opentripplanner/middleware/models/TripMonitorNotification.java]:
```suggestion
                : "Your itinerary is no longer possible on any monitored day of the week. Please plan and save a new trip."
```

File: src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java [hunk 3]
```diff
@@ -423,10 +424,13 @@ private void sendNotifications() {
             LOG.info("Last notifications match current ones. Skipping notify.");
             return;
         }
+
         Map<String, Object> templateData = Map.of(
             "tripId", trip.id,
             "tripName", trip.tripName,
             "notifications", notifications.stream()
+                // Make certain notifications, such as the initial reminder one, appear on top.
+                .sorted(Comparator.comparingInt(TripMonitorNotification::sortOrder))
```
Comment (AdrianaCeric) [inline src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java:433]:
```suggestion
                .sorted(Comparator.comparingInt(notification -> notification.type == NotificationType.INITIAL_REMINDER ? -100 : 0))

```

**Review by AdrianaCeric — APPROVED** [review]
LGTM, few comments

File: src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java (hunk 3 above)
Comment (binh-dam-ibigroup) [reply src/main/java/org/opentripplanner/middleware/tripmonitor/jobs/CheckMonitoredTrip.java:433]: I will keep using the `sortOrder` method to hide the sorting details outside of this class.

**Review by JymDyerIBI — APPROVED** [review]
👍  Looks good.

**Pull request merged**